### ISSUE TITLE
Add contextual help explaining what a "Featured image" is

### ISSUE THREAD

**Issue by simison** [WordPress/gutenberg#30904]
## What problem does this address?
At page/post settings, there's featured image section:

<img width="280" alt="Screenshot 2021-04-16 at 13 08 28" src="https://user-images.githubusercontent.com/87168/115009404-efd5cc00-9eb4-11eb-866d-f368aa5dd173.png">

We don't explain what this does, or why is it important.

## What is your proposed solution?

Add some explanation and link to the documentation.

[AMP plugin](https://wordpress.org/plugins/amp/) actually does add a small explanation, but we could do this directly in Gutenberg. (cc @westonruter).

Here's screenshot with AMP plugin enabled:

<img width="283" alt="Screenshot 2021-04-16 at 13 08 18" src="https://user-images.githubusercontent.com/87168/115009582-1c89e380-9eb5-11eb-80e4-ac73eafffe93.png">

Similar to https://github.com/WordPress/gutenberg/issues/30872

**Comment by karmatosed** [comment]
@simison could you put the messaaging 'in' the block itself that says 'set featured image'? I wonder if there's an option here to combine both together over have 2 areas of text?

**Comment by mtias** [comment]
That text doesn't seem to add anything regarding why it's recommended?

**Comment by simison** [comment]
@mtias I agree, text needs to be something better and/or link to more in-depth docs. That was just an example of AMP plugin adding that exact text currently.

**Comment by westonruter** [comment]
Yes, the text should be better. It is too vague. In the AMP plugin context, the featured image is used in the Schema.org metadata that it adds to the page. So this will ensure that the post will have the expected image when it shows up in Search or when shared on social media. Jetpack and Yoast similarly use the featured image to determine what is included in the metadata for search/social. 

However, since WordPress core doesn't currently include Schema.org metadata generation, I'm not sure that such a notice would be worthwhile at the moment. 

cc @adamsilverstein who has been thinking about Schema.org in core.

**Comment by joedolson** [comment]
At the moment, I'm not sure what core can say about what benefits you can get from setting the featured image - does core, practically speaking, have a way of knowing how the featured image might be used?

**Comment by westonruter** [comment]
No. Not without rendering the theme's singular template for the given post and checking whether the `post_thumbnail_html` filter applies, for example. But even that wouldn't catch cases where the theme grabs the image via `get_post_thumbnail_id()` and renders it using `wp_get_attachment_image()`. 

In practice, though, all themes will display the featured image prominently at the top of the post. So if a user attempts to add an image block as the first block in the content, this seems like an opportunity where the featured image could be mentioned somehow. Like, if a user adds a content-initial image it could ask if they want to use it as a featured image instead, and then explain what the difference is. Maybe it could even be transformed from a block into a featured image at that point. For non-FSE themes, the UI for setting the featured image would seem better under the title as a Post Featured Image block.

**Comment by carolinan** [comment]
The current featured image block behavior is strange, because you can add a featured image block even without the theme_support. It is the theme_support that adds the featured image section to the post setting sidebar.

This section is (only) useful when you want a featured image in for example a query/loop but not the post content.
If you only want the featured image in the content, then you don't need to use this setting, you just upload an image when you place the block.
So perhaps the text should include that information.
"Set a featured image if you want to display the image outside the content"

**Comment by skorasaurus** [comment]
ref https://github.com/WordPress/gutenberg/issues/27617

**Comment by altivero** [comment]
How about:
"This image represents this post or page usually in thumbnail form in a list of posts or pages"
This would cover the most used case and give new users a good example to visualize.

**Comment by bph** [comment]
I would add "The Featured image is also used when your post is shared on social media, i.e. Twitter, Facebook or LinkedIn".

**Comment by ianstewart** [comment]
Is there a great WordPress.org page that explains featured images and why one would want to use them from a writing/publishing perspective?

There's a start of an explanation in the themer/dev-oriented theme handbook.

https://developer.wordpress.org/themes/functionality/featured-images-post-thumbnails/

I don't think that page is the best to link to from the editor but if we take the intro paragraph …

> Featured images (also sometimes called Post Thumbnails) are images that represent an individual Post, Page, or Custom Post Type. When you create your Theme, you can output the featured image in a number of different ways, on your archive page, in your header, or above a post, for example.

That has a good start.

Featured images are images that represent an individual Post, Page, or Custom Post Type. Themes can output the featured image in a number of different ways, on your Blog page, in your header, or above a post, and when your post is shared on social media, like Twitter, Facebook or LinkedIn. Read more about **Featured Images**.

**Comment by bph** [comment]
That's a good point @ianstewart  I don't think there is a page for content creators that explains Featured Images in more detail... [I only found one reference](https://wordpress.org/support/article/settings-sidebar/#featured-image) and that pretty much in the same context, also not explanation why it's important<g>. All other references were in context of individual themes.
[WordPress.com documentation does this better]( https://wordpress.com/support/featured-images/ )

**Comment by noisysocks** [comment]
The popover that https://github.com/WordPress/gutenberg/pull/42352 introduces gives us a natural place to place help text like this cc. @javierarce.

<img width="549" alt="Screen Shot 2022-07-15 at 12 38 01" src="https://user-images.githubusercontent.com/612155/179135981-d3b38fa1-543f-4e17-9008-add3f4a6cfd7.png">

**Comment by joyously** [comment]
This issue is supposed to handle 12742, which is about the existing PHP filter for this that doesn't work in the block editor.
